google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracer.java

```diff
@@ -239,26 +272,43 @@ private void recordOperationCompletion(@Nullable Throwable status) {
       return;
     }
     operationTimer.stop();
+
+    boolean isStreaming = operationType == OperationType.ServerStreaming;
+    String statusStr = Util.extractStatus(status);
+
+    Attributes attributes =
+        baseAttributes
+            .toBuilder()
+            .put(TABLE_ID, tableId)
+            .put(CLUSTER_ID, cluster)
+            .put(ZONE_ID, zone)
+            .put(METHOD, spanName.toString())
+            .put(CLIENT_NAME, NAME)
+            .build();
+
     long operationLatency = operationTimer.elapsed(TimeUnit.MILLISECONDS);
     long operationLatencyNano = operationTimer.elapsed(TimeUnit.NANOSECONDS);
 
     // Only record when retry count is greater than 0 so the retry
     // graph will be less confusing
     if (attemptCount > 1) {
-      recorder.putRetryCount(attemptCount - 1);
+      retryCounter.add(attemptCount - 1, attributes.toBuilder().put(STATUS, statusStr).build());
     }
 
     // serverLatencyTimer should already be stopped in recordAttemptCompletion
-    recorder.putOperationLatencies(operationLatency);
-    recorder.putApplicationLatencies(
-        Duration.ofNanos(operationLatencyNano - totalServerLatencyNano.get()).toMillis());
+    operationLatenciesHistogram.record(
+        operationLatency,
+        attributes.toBuilder().put(STREAMING, isStreaming).put(STATUS, statusStr).build());
+    applicationBlockingLatenciesHistogram.record(
+        Duration.ofNanos(operationLatencyNano - totalServerLatencyNano.get()).toMillis(),
+        attributes);
```

Comment:
not sure if this is coming from the formatter, but this is hard to read, can we add some new lines?